src/main/java/org/example/springproject/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,54 @@
+package org.example.springproject.repository;
+
+import java.util.List;
+import org.example.springproject.model.Book;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```

Comment:
Don't need to use `@Autowired` if you have only one constructor, let's add `@RequiredArgsConstructor` under the class